VikingQuiz.Api/Controllers/QuizzesController.cs

```diff
@@ -23,59 +26,170 @@ public QuizzesController(QuizRepository quizRepository, IEntityMapper<QuizViewMo
             this.vmToEntityMapper = vmToEntityMapper;
             this.entityToVmMapper = entityToVmMapper;
         }
+    
+        private IActionResult FileValidityChecker(NewQuizViewModel quizBodyData)
+        {
+            IActionResult statusCodeResult = Ok();
+            statusCodeResult = Ok();
+            if (quizBodyData.Files.Count == 0) {
+                statusCodeResult =  BadRequest("An image file need to be provided."); }
+            else if (quizBodyData.Files.Count > 1) {
+                statusCodeResult = BadRequest("Only one file can be uploaded.");
+            }
+
+            long size = quizBodyData.Files.Sum(f => f.Length);
+            if (size <= 0) {
+                statusCodeResult = BadRequest("File must not be empty.");
+            }
+
+            var contentType = quizBodyData.Files[0].ContentType;
+            if (!(contentType == "image/gif" || contentType == "image/png" || contentType == "image/jpeg")) {
+                statusCodeResult =  BadRequest("Only images of the following formats are allowed: .png, .jpeg or .gif");
+            }
 
+            return statusCodeResult;
+        }
+
+
+        // Get all the quizzes which belong to an user
         [HttpGet]
-        public IActionResult GetAll()
+        //[Authorize] -- add authorize when it is available on the front end
+        public async Task<IActionResult> GetAll()
         {
-            var quizzes = quizRepository.GetAll();
-            return Ok(quizzes.Select(quiz => this.entityToVmMapper.Map(quiz)));
+            // User.Claims.GetUserId(); -- use this to get the ID of the user based on the token recieved
+
+            AzureBlobService blobService = new AzureBlobService();
+            await blobService.InitializeBlob();
+
+            var quizzes = quizRepository.GetAll(3); // pass that user ID in
```

Comment:
Please remove the comment and change the parameter passed to that function. We should avoid sending magic numbers in method calls.